db/active_replicator_push.go

```diff
@@ -88,6 +90,22 @@ func (apr *ActivePushReplicator) _connect() error {
 		channels = base.SetFromArray(apr.config.FilterChannels)
 	}
 
+	// This should always be set to true in a real scenario. This is only to be used for tests
+	ignoreNoConflicts := true
+	if apr.config.DisableIgnoreNoConflicts {
+		ignoreNoConflicts = false
+	}
+
+	apr.blipSyncContext.fatalErrorCallback = func(err error) {
```

Comment:
Having the blipSyncContext send fatalError notification up to the replicator, and then relying on the (external) replicator to turn around and close the blipSyncContext (in stopAndDisconnect) doesn't seem like the ideal design.  
I think it would be preferable for blipSyncContext to close itself, set an error state and then have the replicator respond to and detect that.

We're already doing something similar for the Complete() handling in _connect - it gets triggered when sendChanges eventually returns.  Having sendChanges return error might be sufficient for communication from the context back to the replicator in this case.

Reply:
As discussed, some type of communication up to the replicator is necessary in order to close any additional replications that are running (e.g. associated pull for a pushAndPull replication).  We should review whether this could be done by expanding the onComplete callback to support sending error information, or more generally if there's a better option for communication from bsc back to consumers.

Reply:
Changes the empty changes callback to be a more generic 'onReplicationComplete' which can take an error.